supervision/dataset/formats/coco.py

```diff
@@ -83,20 +97,33 @@ def coco_annotations_to_detections(
     xyxy[:, 2:4] += xyxy[:, 0:2]
 
     if with_masks:
-        polygons = [
-            np.reshape(
-                np.asarray(image_annotation["segmentation"], dtype=np.int32), (-1, 2)
-            )
-            for image_annotation in image_annotations
-        ]
-        mask = _polygons_to_masks(polygons=polygons, resolution_wh=resolution_wh)
+        mask = coco_annotations_to_masks(
+            image_annotations=image_annotations, resolution_wh=resolution_wh
+        )
         return Detections(
             class_id=np.asarray(class_ids, dtype=int), xyxy=xyxy, mask=mask
         )
 
     return Detections(xyxy=xyxy, class_id=np.asarray(class_ids, dtype=int))
 
 
+def _mask_has_holes(mask: np.ndarray) -> bool:
```

Comment:
Let's move the `_mask_has_holes` and `_mask_has_multiple_segments` functions to [`supervision/detection/utils.py`](https://github.com/roboflow/supervision/blob/develop/supervision/detection/utils.py) and make them publicly accessible. These features are valuable utilities for Supervision users and should be prominently featured.

Additionally, please:

- Add these functions to the main [`__init__.py`](https://github.com/roboflow/supervision/blob/develop/supervision/__init__.py) file to make them available in the public API.
- Include comprehensive docstrings for each function to explain their use and parameters.
- Document these functions in [`utils.md`](https://github.com/roboflow/supervision/blob/develop/docs/detection/utils.md) to ensure they are visible in our documentation.
- Add unit tests for both functions.

Reply:
resolved with commits from 2626263ff3f552634b301501b4824df160f2ee47 to 6dbf3f61e3e764c56c6152f18088f7624354d1dd